### ISSUE TITLE
Runtime Error: No native client found

### ISSUE THREAD

**Issue by beausmith** [bugsnag/bugsnag-react-native#269]
## Description
The `ios/kip.xcodeproj/project.pbxproj` file in my existing project was not updated when I ran `react-native link bugsnag-react-native`.

## Issue
I am adding `bugsnag-react-native` to an existing project with other 3rd party libraries including Cocoapod files.

I installed using:

```
yarn add bugsnag-react-native
react-native link bugsnag-react-native
```

Setup basic client:

```
import { Client } from 'bugsnag-react-native'
const bugsnag = new Client('00f4edda951798ef704d19fa41911d86')
bugsnag.notify(Error(`testing ${new Date().toString()}`))
```

I run `react-native run-ios` and get this error:

> Bugsnag: No native client found. Is BugsnagReactNative installed in your native code project?

After installing `bugsnag-react-native` in a clean react-native project, I noticed that the `ios/kip.xcodeproj/project.pbxproj` file in my existing project was not updated when I ran `react-native link bugsnag-react-native`.

After some tinkering I found that the existence of Cocoapod files was causing the issue.

## Solution

To remedy this issue, I found that removing the Pod* files before install and then replacing them after allowed for the basic linking to work.

```
rm -rf ios/Pod*
yarn add bugsnag-react-native
react-native link bugsnag-react-native
git co -- ios/Pod*
```

## Related 

https://github.com/bugsnag/bugsnag-react-native/issues/14

**Comment by fractalwrench** [comment]
Thanks for the detailed report @beausmith - we'll look into this in a bit more depth and see whether this is something we need to workaround or document.

**Comment by tombeynon** [comment]
Thank you so much @beausmith - this was the only solution we'd found to the Cocoapods/React Native install hell. We were unable to install Bugsnag using Cocoapods, so had removed it from the Podfile, but linking Bugsnag then became impossible until we found this issue.

**Comment by isaaclem** [comment]
I'm facing the problem which I think is due to the POD file too. I'm wondering what are the options available to solve this problem?

**Comment by nihp** [comment]
@kattrali Same issue here.

 While I am trying to add the BugsnagReactNative.Xcodeproj, I can't able to view this under node_modules.

**Comment by bradleyflood** [comment]
I also had to 
```
cd ios
pod install
```
After this to get it working :)

**Comment by thecodecafe** [comment]
I'm getting this same error and I'm using the following pair.

**bugsnag-react-native: "^2.23.1",
react-native: "0.60.5",**
Platform: iOS

Have reinstalled node_modules and pods several times, but nothing changed.

**Comment by mattdyoung** [comment]
@thecodecafe 
What are the full details of the error you're seeing?

I can see you also commented on https://github.com/bugsnag/bugsnag-react-native/issues/377#issuecomment-534602440 as well.

Are you able to share a reproduction example that we try out?

**Comment by thecodecafe** [comment]
# Fix For When Upgrading To RN 0.60.5
### You probably missed some things in the Podfile.

So I was able to fix this, turns out the problem was from me when I upgraded to `RN 0.60.5` I didn't do the following.
Within the **Podfile** add the following right underneath the `platform:ios` line
```
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
```
Then right before the `end` of your main target podspecs list section add the following line.
```
use_native_modules!
```
Now if you run `pod install` in the ios directory in your terminal it will install the pods for all your installed third-party libraries in `node_modules` as well as that of `bugsnag-react-native`.
This will tell react to look into the `node_modules` folder and collate all the pod specs available for the libraries you installed, it will use that to install the pods for you when you run `pod install` in the ios folder.

**NB:** Don't forget to run `react-native unlink bugsnag-react-native` if you had it linked before.

Sorry for the troubles @mattdyoung already found a solution.

**Comment by abigail-bugsnag** [comment]
@thecodecafe Thanks! We are looking into improvements in our documentation in this area.
